### ISSUE TITLE
Incorrect "static anonymous functions cannot access $this" due to nesting

### ISSUE THREAD

**Issue by reedy** [hammer-tools/php-hammer#21]
With a nested class inside a `static function ()`, it is possible that usages of `$this` are actually valid

https://github.com/wikimedia/mediawiki/blob/8a9dd67/tests/phpunit/includes/specialpage/SpecialPageFactoryTest.php#L348-L369

```php
		$this->overrideConfigValue( MainConfigNames::SpecialPages, [
			'ExceptionPage' => [
				'factory' => static function () use ( $expectedException ) {
					return new class( $expectedException ) extends SpecialPage {
						private Exception $expectedException;

						public function __construct( $expectedException ) {
							parent::__construct();
							$this->expectedException = $expectedException;
						}

						public function execute( $par ) {
							throw $this->expectedException;
						}

						public function isIncludable() {
							return true;
						}
					};
				},
			]
		] );
```

**Comment by rentalhost** [comment]
Thanks!

**Comment by rentalhost** [comment]
Thank you for taking the time to report this issue. The fix has been implemented in version **0.24.0** and is available on the **[Releases](https://github.com/hammer-tools/php-hammer/releases)** page for manual installation. Alternatively, you can wait for the official release on the **JetBrains Marketplace**, which should happen in approximately 3 days.

- **Regarding the issue you mentioned:** Now, the `$this` variable will be checked immediately within the current function to which it belongs. This means that even when nested, you can use `static` in an "outer" scope without any problem.
